uvtools/tests/test_dspec.py

```diff
@@ -797,6 +808,17 @@ def get_snr(clean, fftax=1, avgax=0, modes=[2, 20]):
 
     #check that fringe rate filter model gives similar results to delay filter.
     nt.assert_true(np.all(np.isclose(mdl1[~f],mdl5[~f], rtol=1e-2)))
+
+    #perform some sanity checks on handling of nans in dft_leastsq
+    mdl, res, info = dspec.fourier_filter(x=freqs, data=d[0], wgts=w[0], filter_centers=[0.],
+                                             filter_half_widths=[bl_len], suppression_factors=[1e-9],
+                                             mode='dft_leastsq')
+    # check that the filter_period is indeed equal to 1 / (2 * bandwidth)
+    nt.assert_true(np.isclose(info['filter_params']['axis_1']['basis_options']['fundamental_period'],
```

Comment:
This doesn't test all cases of `dft_leastsq` handling the fundamental period. It tests that the default fundamental period is set appropriately, but it doesn't test the case where the user provides a fundamental period different than the default setting. This second part is important to test, since this test would *not* catch the bug I pointed out in a previous comment (where the code was setting the fundamental period to the default regardless of the input it was provided).

Reply:
I've added a test. Also check 2d.